Flakey Test: SiteCommandIT ->  test_Command_Site_Pull_Pull_All_Default_Format

### Parent Issue

_No response_

### Problem Statement

The test failed on [this run](https://github.com/dotCMS/core/actions/runs/10477824201/job/29020314042)

com.dotcms.cli.command.site.SiteCommandIT ► Test_Command_Site_Pull_Pull_All_Default_Format

```
Failed test found in:
  /tmp/build-reports-test/build-reports-test-CLI Tests/tools/dotcms-cli/cli/target/failsafe-reports/TEST-com.dotcms.cli.command.site.SiteCommandIT.xml
Error:
  org.opentest4j.AssertionFailedError: expected: <0> but was: <1>
```

More log details:
```
org.opentest4j.AssertionFailedError: expected: <0> but was: <1>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:527)
	at com.dotcms.cli.command.site.SiteCommandIT.Test_Command_Site_Pull_Pull_All_Default_Format(SiteCommandIT.java:643)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:1013)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:827)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	Suppressed: com.dotcms.api.client.pull.exception.PullException: RESTEASY003145: Unable to find a MessageBodyReader of content-type application/octet-stream and type class com.dotcms.model.ResponseEntityView
		at com.dotcms.api.client.util.ErrorHandlingUtilImpl.mapException(ErrorHandlingUtilImpl.java:198)
		at com.dotcms.api.client.util.ErrorHandlingUtilImpl.mapPullException(ErrorHandlingUtilImpl.java:175)
		at com.dotcms.api.client.util.ErrorHandlingUtilImpl_ClientProxy.mapPullException(Unknown Source)
		at com.dotcms.api.client.pull.PullServiceImpl.fetch(PullServiceImpl.java:151)
		at com.dotcms.api.client.pull.PullServiceImpl.pull(PullServiceImpl.java:48)
		at com.dotcms.api.client.pull.PullServiceImpl_Subclass.pull$$superforward(Unknown Source)
		at com.dotcms.api.client.pull.PullServiceImpl_Subclass$$function$$1.apply(Unknown Source)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:124)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
		at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
		at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
		at com.dotcms.api.client.pull.PullServiceImpl_Subclass.pull(Unknown Source)
		at com.dotcms.cli.command.site.SitePull.pull(SitePull.java:100)
		at com.dotcms.cli.command.site.SitePull.call(SitePull.java:82)
		at com.dotcms.cli.command.site.SitePull_Subclass.call$$superforward(Unknown Source)
		at com.dotcms.cli.command.site.SitePull_Subclass$$function$$5.apply(Unknown Source)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:124)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
		at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
		at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
		at com.dotcms.cli.command.site.SitePull_Subclass.call(Unknown Source)
		at com.dotcms.cli.command.site.SitePull.call(SitePull.java:24)
		at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
		at picocli.CommandLine.access$1500(CommandLine.java:148)
		at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
		at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
		at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
		at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
		at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
		at com.dotcms.cli.common.DotExecutionStrategy.internalExecute(DotExecutionStrategy.java:183)
		at com.dotcms.cli.common.DotExecutionStrategy.execute(DotExecutionStrategy.java:94)
		at picocli.CommandLine.execute(CommandLine.java:2170)
		at com.dotcms.cli.command.site.SiteCommandIT.Test_Command_Site_Pull_Pull_All_Default_Format(SiteCommandIT.java:641)
		... 5 more
	Caused by: jakarta.ws.rs.ProcessingException: RESTEASY003145: Unable to find a MessageBodyReader of content-type application/octet-stream and type class com.dotcms.model.ResponseEntityView
		at org.jboss.resteasy.core.interception.jaxrs.ClientReaderInterceptorContext.throwReaderNotFound(ClientReaderInterceptorContext.java:46)
		at org.jboss.resteasy.core.interception.jaxrs.AbstractReaderInterceptorContext.getReader(AbstractReaderInterceptorContext.java:129)
		at org.jboss.resteasy.core.interception.jaxrs.AbstractReaderInterceptorContext.proceed(AbstractReaderInterceptorContext.java:76)
		at org.jboss.resteasy.client.jaxrs.internal.ClientResponse.readFrom(ClientResponse.java:192)
		at org.jboss.resteasy.specimpl.BuiltResponse.readEntity(BuiltResponse.java:75)
		at org.jboss.resteasy.specimpl.AbstractBuiltResponse.readEntity(AbstractBuiltResponse.java:227)
		at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.extractResult(ClientInvocation.java:164)
		at org.jboss.resteasy.client.jaxrs.internal.proxy.extractors.BodyEntityExtractor.extractEntity(BodyEntityExtractor.java:59)
		at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invokeSync(ClientInvoker.java:136)
		at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:103)
		at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:61)
		at com.sun.proxy.$Proxy121.findByName(Unknown Source)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at io.quarkus.restclient.runtime.QuarkusProxyInvocationHandler.invoke(QuarkusProxyInvocationHandler.java:165)
		at com.sun.proxy.$Proxy122.findByName(Unknown Source)
		at com.dotcms.api.client.pull.site.SiteFetcher.fetchByKey(SiteFetcher.java:96)
		at com.dotcms.api.client.pull.site.SiteFetcher_Subclass.fetchByKey$$superforward(Unknown Source)
		at com.dotcms.api.client.pull.site.SiteFetcher_Subclass$$function$$1.apply(Unknown Source)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:124)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
		at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
		at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
		at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
		at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
		at com.dotcms.api.client.pull.site.SiteFetcher_Subclass.fetchByKey(Unknown Source)
		at com.dotcms.api.client.pull.site.HttpRequestTask.lambda$compute$0(HttpRequestTask.java:67)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
		at java.base/java.util.ArrayList$SubList$2.forEachRemaining(ArrayList.java:1511)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
		at com.dotcms.api.client.pull.site.HttpRequestTask.lambda$compute$1(HttpRequestTask.java:68)
		at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
		at io.smallrye.context.impl.wrappers.SlowContextualRunnable.run(SlowContextualRunnable.java:19)
		at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:587)
		at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
		at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
		at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
		at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base/java.lang.Thread.run(Thread.java:829)
```

### Steps to Reproduce

Flakey test

### Acceptance Criteria

This test should always pass

### dotCMS Version

trunk

### Proposed Objective

Code Maintenance

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/site/SiteCommandIT.java
+++ b/tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/site/SiteCommandIT.java
@@ -18,6 +18,8 @@ import com.dotcms.model.site.SiteVariableView;
 import com.dotcms.model.site.SiteView;
 import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.junit.TestProfile;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.NotFoundException;
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
@@ -30,8 +32,6 @@ import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import jakarta.inject.Inject;
-import jakarta.ws.rs.NotFoundException;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
@@ -619,30 +619,21 @@ class SiteCommandIT extends CommandTest {
 
             // ---
             // Creating some test sites
-            final String newSiteName1 = String.format("new.dotcms.site1-%d",
-                    System.currentTimeMillis());
-            var status = commandLine.execute(SiteCommand.NAME, SiteCreate.NAME, newSiteName1);
-            Assertions.assertEquals(CommandLine.ExitCode.OK, status);
+            sitesTestHelper.createSiteOnServer().siteName();
             sitesCount++;
 
-            final String newSiteName2 = String.format("new.dotcms.site2-%d",
-                    System.currentTimeMillis());
-            status = commandLine.execute(SiteCommand.NAME, SiteCreate.NAME, newSiteName2);
-            Assertions.assertEquals(CommandLine.ExitCode.OK, status);
+            sitesTestHelper.createSiteOnServer().siteName();
             sitesCount++;
 
-            final String newSiteName3 = String.format("new.dotcms.site3-%d",
-                    System.currentTimeMillis());
-            status = commandLine.execute(SiteCommand.NAME, SiteCreate.NAME, newSiteName3);
-            Assertions.assertEquals(CommandLine.ExitCode.OK, status);
+            sitesTestHelper.createSiteOnServer().siteName();
             sitesCount++;
 
             // Pulling all sites
-            status = commandLine.execute(SiteCommand.NAME, SitePull.NAME,
+            final var status = commandLine.execute(SiteCommand.NAME, SitePull.NAME,
                     "--workspace", workspace.root().toString());
             Assertions.assertEquals(CommandLine.ExitCode.OK, status);
 
-            // Make sure we have the proper amount of JSON files in the sites folder
+            // Make sure we have the proper number of JSON files in the sites folder
             try (Stream<Path> walk = Files.walk(workspace.sites())) {
 
                 var jsonFiles = walk.filter(Files::isRegularFile)

